aioshelly/coap.py

```diff
@@ -1,18 +1,27 @@
 """COAP for Shelly."""
 import asyncio
+import ipaddress
 import json
 import logging
 import socket
 import struct
+import time
 from typing import Optional, cast
 
 import netifaces
-
-_LOGGER = logging.getLogger(__name__)
+from scapy.contrib.igmp import IGMP
```

Comment:
Scapy requires root I think?

Reply:
I don't need any special permission to run inside HA.